### PR TITLE
[1LP][RFR] Added new update tests and fixtures for them

### PR DESCRIPTION
**Pull request by lcouzens**
Added new update tests for different scenarios and moved all fixtures to cli.py for central usage.

REQUIRES PR: 7759

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 718dd14a728ed1f84037ba87ffe0c22fa8df1625:

`cfme/fixtures/cli.py`:
- :bangbang: Line 497:1: **E305** *expected 2 blank lines after class or function definition, found 0*
- :red_circle: Line 497:1: **F821** *undefined name 'coo'*
- :heavy_exclamation_mark: Line 497:4: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/fixtures/cli.py [hunk 1]
```diff
@@ -1,17 +1,26 @@
-import pytest
-
 from collections import namedtuple
 from contextlib import contextmanager
+
+import fauxfactory
+import pytest
 from six import iteritems
 
 import cfme.utils.auth as authutil
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: sort imports alphabetically please .
This can move to line 15

File: cfme/fixtures/cli.py [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
+from cfme.cloud.provider.ec2 import EC2Provider
+from cfme.configure.configuration.region_settings import RedHatUpdates
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.test_framework.sprout.client import SproutClient, SproutException
+from cfme.utils import conf
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.conf import credentials, auth_data, cfme_data
 from cfme.utils.log import logger
+from cfme.utils.providers import list_providers_by_class
+from cfme.utils.version import Version
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Please remove the line below and use
from cfme.utils.version import get_stream , Version

File: cfme/fixtures/cli.py [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
 from cfme.utils.version import get_stream
 from cfme.utils.wait import wait_for
 
-TimedCommand = namedtuple('TimedCommand', ['command', 'timeout'])
+REPOSITORIES = ["https://github.com/lcouzens/ansible_playbooks"]
+TimedCommand = namedtuple("TimedCommand", ["command", "timeout"])
 
 """ The Following fixtures are for provisioning one preconfigured or unconfigured appliance for
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Please move the docstring to respective fixtures

File: cfme/fixtures/cli.py [hunk 4]
```diff
@@ -64,8 +73,8 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
     '1' selects internal db, '1' use partition, 'y' create dedicated db, 'pwd'
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Please format the docstring .
Example :  https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/cli/test_appliance_console.py#L59

File: cfme/fixtures/cli.py [hunk 5]
```diff
@@ -74,47 +83,414 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
...
-        'sshlogin': credentials['ssh']['username'],
-        'sshpass': credentials['ssh']['password']
+        "username": credentials["database"]["username"],
+        "password": credentials["database"]["password"],
+        "sshlogin": credentials["ssh"]["username"],
+        "sshpass": credentials["ssh"]["password"],
     }
+
+
+def temp_appliances(
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Required :  
        def temp_appliances(count=1, preconfigured=True, 
                            lease_time=180, stream=None, 
                            provider_type=None):

File: cfme/fixtures/cli.py [hunk 6]
```diff
@@ -74,47 +83,414 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
...
+    apps0, apps1, apps2 = ha_multiple_preupdate_appliances
+    app0_ip = apps0.hostname
+    app1_ip = apps1.hostname
+    pwd = app_creds["password"]
+    # Configure first appliance as dedicated database
+    command_set = ("ap", "", "5", "1", "1", "1", "y", pwd, TimedCommand(pwd, 360), "")
+    apps0.appliance_console.run_commands(command_set)
+    wait_for(lambda: apps0.db.is_dedicated_active)
+    # Configure EVM webui appliance with create region in dedicated database
+    command_set = (
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Required : 
        
    command_set = ("ap",  "",  "5",  "2", app0_ip, "", pwd, "", "2",  "0",  "y", 
                   app0_ip, "",  "",  "",  pwd, TimedCommand(pwd, 360), "" )

File: cfme/fixtures/cli.py [hunk 7]
```diff
@@ -74,47 +83,414 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
...
+        "",
+        pwd,
+        TimedCommand(pwd, 360),
+        "",
+    )
+    apps2.appliance_console.run_commands(command_set)
+    apps2.wait_for_evm_service()
+    apps2.wait_for_web_ui()
+    # Configure primary replication node
+    command_set = (
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Same for all command sets , Please use single line separated by commas

File: cfme/fixtures/cli.py [hunk 8]
```diff
@@ -74,47 +83,414 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
...
+        pwd,
+        pwd,
+        app0_ip,
+        "y",
+        TimedCommand("y", 60),
+        "",
+    )
+    apps0.appliance_console.run_commands(command_set)
+    # Configure secondary replication node
+    command_set = (
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: Same as above

File: cfme/utils/appliance/__init__.py [hunk 9]
```diff
@@ -131,6 +134,53 @@ def run_commands(self, commands, autoreturn=True, timeout=10, channel=None):
                 pass
             logger.debug(result)
 
+    def scap_harden_appliance(self):
+        """'ap' launches appliance_console, '' clears info screen, '14' Hardens appliance using SCAP
```
Comment (sshveta) [inline cfme/utils/appliance/__init__.py]: Please format docstring here too

File: cfme/utils/appliance/__init__.py [hunk 10]
```diff
@@ -131,6 +134,53 @@ def run_commands(self, commands, autoreturn=True, timeout=10, channel=None):
...
+    def scap_check_rules(self):
+        """Check that rules have been applied correctly."""
+        rules_failures = []
+        with tempfile.NamedTemporaryFile('w') as f:
+            f.write(hidden['scap.rb'])
+            f.flush()
+            os.fsync(f.fileno())
+            self.appliance.ssh_client.put_file(
+                f.name, '/tmp/scap.rb')
+        if self.appliance.version >= "5.8":
```
Comment (sshveta) [inline cfme/utils/appliance/__init__.py]: version 5.8 check can be removed now

File: cfme/fixtures/cli.py [hunk 11]
```diff
@@ -74,47 +83,414 @@ def dedicated_db_appliance(app_creds, unconfigured_appliance):
...
+
+
+@pytest.fixture(scope="module")
+def ansible_repository(appliance):
+    repositories = appliance.collections.ansible_repositories
+    repository = repositories.create(
+        fauxfactory.gen_alpha(), REPOSITORIES[0], description=fauxfactory.gen_alpha()
+    )
+    view = navigate_to(repository, "Details")
+    if appliance.version < "5.9":
```
Comment (sshveta) [inline cfme/fixtures/cli.py]: No need to check for version less than 5.9 .
We are only doing 5.9 and 5.10 now

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/fixtures/cli.py (hunk 1 above)
Comment (lcouzens) [reply cfme/fixtures/cli.py]: I believe this is the correct ordering as its 'import' not 'from'

File: cfme/utils/appliance/__init__.py (hunk 10 above)
Comment (lcouzens) [reply cfme/utils/appliance/__init__.py]: Required for upgrade testing.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by sshveta** [comment]
Running PRT again .

**Comment by sshveta** [comment]
@lcouzens :please comment on 5.9 failures 

     logger.error("Couldn't provision appliance with following error:{}".format(e))
>           raise SproutException("Not able to configure provision request")
E           SproutException: Not able to configure provision request

**Comment by lcouzens** [comment]
Running again, but looks like errors are from sprout not being able to provide the provisions in time.

**Comment by lcouzens** [comment]
All failures are from sprout provisions.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by sshveta** [comment]
Rebase again pls

**Pull request merged**